nucypher/blockchain/eth/agents.py

```diff
@@ -601,6 +601,20 @@ def set_restaking(self, transacting_power: TransactingPower, value: bool) -> TxR
         # TODO: Handle ReStakeSet event (see #1193)
         return receipt
 
+    @contract_api(CONTRACT_CALL)
+    def is_migrated(self, staker_address: ChecksumAddress) -> bool:
+        flags = self.get_flags(staker_address)
+        return flags.migration_flag
+
+    @contract_api(TRANSACTION)
+    def migrate(self, transacting_power: TransactingPower, staker_address: ChecksumAddress = None) -> TxReceipt:
```

Comment:
Should `staker_address` be Optional?

```suggestion
    def migrate(self, transacting_power: TransactingPower, staker_address: Optional[ChecksumAddress] = None) -> TxReceipt:
```

Reply:
✔️